ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp

```diff
@@ -267,9 +272,20 @@ Tensor all_gather(
             const std::vector<Tensor>& input_tensors,
             const std::vector<std::optional<const Tensor>>& optional_input_tensors,
             const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
-            const auto& input_tensor = input_tensors.at(0);
-
-            return operation::run(
+            auto input_tensor = input_tensors.at(0);
+
+            bool needs_padding = false;
+            uint32_t total_elements = input_tensor.get_logical_shape()[dim];
+            if (input_tensor.get_logical_shape()[-2] < tt::constants::TILE_HEIGHT ||
```

Comment:
This can be cleaned up a little but also the condition should be expanded to also include shapes that are > TILE size but still not tile aligned (so they will still have padding)
```suggestion
            bool needs_padding = input_tensor.get_logical_shape()[-2] % tt::constants::TILE_HEIGHT != 0 || input_tensor.get_logical_shape()[-1] % tt::constants::TILE_WIDTH != 0;
```